evap/staff/tests/test_views.py

```diff
@@ -1923,6 +1923,102 @@ def test_evaluation_copy(self):
+        # Default the evaluation does not have any participants, so we need to add some
+        cls.evaluation.participants.set(baker.make(UserProfile, _quantity=3))
+
+    def test_create_exam_evaluation(self):
+        self.app.post(
+            self.url,
+            user=self.manager,
+            status=200,
+            params={"evaluation_id": self.evaluation.pk, "exam_date": self.exam_datetime},
+        )
+        self.assertEqual(Evaluation.objects.count(), 2)
+        exam_evaluation = Evaluation.objects.exclude(pk=self.evaluation.pk).get()
+        self.assertEqual(exam_evaluation.contributions.count(), self.evaluation.contributions.count())
+        self.assertEqual(
+            exam_evaluation.vote_start_datetime,
+            datetime.datetime.combine(self.exam_datetime + datetime.timedelta(days=1), datetime.time(8, 0)),
+        )
+        self.assertEqual(exam_evaluation.vote_end_date, self.exam_datetime + datetime.timedelta(days=3))
+        self.assertEqual(exam_evaluation.name_de, "Klausur")
+        self.assertEqual(exam_evaluation.name_en, "Exam")
+        self.assertEqual(exam_evaluation.course, self.evaluation.course)
+        self.assertQuerySetEqual(exam_evaluation.participants.all(), self.evaluation.participants.all())
+        self.assertEqual(exam_evaluation.weight, 1)
+
+        evaluation = Evaluation.objects.get(pk=self.evaluation.pk)
+        self.assertEqual(evaluation.weight, 9)
+
+    def test_exam_evaluation_for_single_result(self):
+        self.evaluation.is_single_result = True
+        self.evaluation.save()
+        self.app.post(
+            self.url,
+            user=self.manager,
+            status=400,
+            params={"evaluation_id": self.evaluation.pk, "exam_date": self.exam_datetime},
+        )
+        self.assertFalse(self.evaluation.has_exam)
+
+    def test_exam_evaluation_for_already_existing_exam_evaluation(self):
+        baker.make(Evaluation, course=self.course, name_en="Exam", name_de="Klausur")
+        self.assertTrue(self.evaluation.has_exam)
+        self.app.post(
+            self.url,
+            user=self.manager,
+            status=400,
+            params={"evaluation_id": self.evaluation.pk, "exam_date": self.exam_datetime},
+        )
+
+    def test_exam_evaluation_with_wrong_date(self):
+        self.evaluation.vote_start_datetime = datetime.datetime.now() + datetime.timedelta(days=100)
+        self.evaluation.vote_end_date = datetime.date.today() + datetime.timedelta(days=150)
+        self.evaluation.save()
+        self.app.get("", user=self.manager)  # Needed to not get a last login database update
+
+        with assert_no_database_modifications():
+            self.app.post(
+                self.url,
+                user=self.manager,
+                status=400,
+                params={"evaluation_id": self.evaluation.pk, "exam_date": self.exam_datetime},
+            )
```

Comment:
The parameters are duplicated across the tests -- this could also all just be one tests that defined all those "base" arguments once -- or maybe have these parameters stored in the class or something